lib/wordcounter.rb

```diff
@@ -0,0 +1,93 @@
+require 'delegate'
+
+class WordHistogram < SimpleDelegator
+  def initialize(*args)
+    super(Hash.new(0, *args))
+  end
+
+  def to_pretty_s
+    word_freq_pairs = self.sort_histogram
+
+    word_lengths = word_freq_pairs.map { |word, _| word }.map &:length
+    max_length = word_lengths.max
+
+    self.to_table_s(word_freq_pairs, middle_column_width: max_length)
+  end
+
+  protected
```

Comment:
You drop the `self.` part of these method calls and make them private. Private methods can be called from within the class if you don't put `self.` on them. `self.` gives you the same access as external callers.

The only time you need `self.` on a method call is to disambiguate it from setting a local variable:

```ruby
def my_attribute=(val)
  p "setting the instance variable"
  @my_attribute = val
end

my_attribute = :foo
defined?(my_attribute)     #=> "local-variable"
@my_attribute              #=> nil

self.my_attribute = :bar
                           #=> "setting the instance variable"
@my_attribute              #=> :bar
```